src/resources/PackageResource.ts

```diff
@@ -0,0 +1,123 @@
+import { Response } from 'node-fetch';
+
+import { ExportedAuditTrail, Package, Requests, Responses } from '../types';
+import { serializeDate } from '../utils/serializeDate';
+import { Resource } from './Resource';
+
+/**
+ * Resource class to interact with
+ * {@link https://community.onespan.com/documentation/onespan-sign/guides/sender-guides/user/transactions | packages (transactions)}.
+ *
+ * @public
+ */
+export class PackageResource extends Resource {
+  /**
+   * Creates a package (transaction).
+   *
+   * @param payload - Package initial information
+   * @returns A payload that contains the ID of the newly created package.
+   *
+   * @remarks
+   * - {@link https://community.onespan.com/products/onespan-sign/sandbox#/Packages/api.packages.post | REST API documentation (OneSpan)}
+   *
+   * - {@link https://community.onespan.com/documentation/onespan-sign/guides/feature-guides/developer/creating-transaction-sender | Creating a Transaction for a Sender (OneSpan)}
+   */
+  public async create(payload: Requests.PackageData): Promise<Responses.CreatePackage> {
+    const response = await this.api.post('/api/packages').withBody(payload).fetch();
+    return (await response.json()) as Responses.CreatePackage;
+  }
+
+  /**
+   * Retrieves all packages (transactions).
+   *
+   * @remarks
+   * - {@link https://community.onespan.com/products/onespan-sign/sandbox#/Packages/api.packages.get | REST API documentation (OneSpan)}
+   *
+   * - {@link https://community.onespan.com/documentation/onespan-sign/guides/feature-guides/developer/retrieving-list-transactions | Retrieving a List of Transactions (OneSpan)}
+   */
+  public async getAll(params: Requests.GetAllPackagesParameters): Promise<Package[]> {
+    const { lastUpdatedEndDate, lastUpdatedStartDate } = params;
+
+    const queryParameters = {
+      ...params,
+      lastUpdatedStartDate: serializeDate(lastUpdatedStartDate),
```

Comment:
why do we need to always specify `lastUpdatedStartDate` and `lastUpdatedEndDate`?

Reply:
`serializeDate` may return `undefined` if `undefined` is passed in. In that case, the property won't be added to the query parameters